test/built-ins/Set/prototype/union/set-like-class-mutation.js

```diff
@@ -0,0 +1,42 @@
+// Copyright (C) 2023 Kevin Gibbons, Anthony Frehner. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: pending
+description: Set.prototype.union maintains values even when a custom Set-like class mutates the receiver
```

Comment:
Nice and evil. I like it.

Note that there are observable calls into user code between the start of the call and the copy of the `[[SetData]]`. Please cover that as well and/or file an issue on the proposal to verify that's intended.

Reply:
It's not exactly intended but I guess not exactly not intended, and given that this is stage 3 I don't consider it worth tweaking. Added a side-effect in one of those places, though I didn't try to be exhaustive.